scopes/workspace/install/install.main.runtime.ts

```diff
@@ -257,9 +255,13 @@ export class InstallMain {
       }
       await this.link(linkOpts);
       prevManifests.add(hash(current.manifests));
+      // We need to clear cache before creating the new component manifests.
+      this.workspace.consumer.componentLoader.clearComponentsCache();
+      this.workspace.clearCache();
       current = await this._getComponentsManifests(installer, mergedRootPolicy, pmInstallOptions);
       installCycle += 1;
     } while ((!prevManifests.has(hash(current.manifests)) || hasMissingLocalComponents) && installCycle < 5);
+    await this.workspace.consumer.componentFsCache.deleteAllDependenciesDataCache();
```

Comment:
maybe I should put it above line 261

Reply:
@davidfirst what do you think about adding this to `workspace.clearCache`? It would make the function async, which is might be a problem as it is called in the sync clearCache in `consumer.ts`

Reply:
we'll discuss this separately